bats/index.bats

```diff
@@ -0,0 +1,1704 @@
+}
+
+@test "index: SELECT = Full Match" {
+    dolt sql <<SQL
+CREATE INDEX idx_v1 ON onepk(v1);
+CREATE INDEX idx_v ON twopk(v2, v1);
+INSERT INTO onepk VALUES (1, 99, 51), (2, 11, 55), (3, 88, 52), (4, 22, 54), (5, 77, 53);
+INSERT INTO twopk VALUES (1, 99, 51, 63), (2, 11, 55, 64), (3, 88, 52, 61), (4, 22, 54, 65), (5, 77, 53, 61);
+SQL
+    # found
+    run dolt sql -q "SELECT * FROM onepk WHERE v1 = 77"
+    [ "$status" -eq "0" ]
+    [[ "${lines[1]}" = "| pk1 | v1 | v2 |" ]] || false
+    [[ "${lines[3]}" = "| 5   | 77 | 53 |" ]] || false
+    [[ "${lines[4]}" = "+-----+----+----+" ]] || false
+    # not found
+    run dolt sql -q "SELECT * FROM onepk WHERE v1 = 111"
+    [ "$status" -eq "0" ]
+    [[ "${lines[1]}" = "| pk1 | v1 | v2 |" ]] || false
+    [[ "${lines[3]}" = "+-----+----+----+" ]] || false
+    # not indexed & found
+    run dolt sql -q "SELECT * FROM onepk WHERE v2 = 54"
+    [ "$status" -eq "0" ]
+    [[ "${lines[1]}" = "| pk1 | v1 | v2 |" ]] || false
+    [[ "${lines[3]}" = "| 4   | 22 | 54 |" ]] || false
+    [[ "${lines[4]}" = "+-----+----+----+" ]] || false
+    # not indexed & not found
+    run dolt sql -q "SELECT * FROM onepk WHERE v2 = 111"
+    [ "$status" -eq "0" ]
+    [[ "${lines[1]}" = "| pk1 | v1 | v2 |" ]] || false
+    [[ "${lines[3]}" = "+-----+----+----+" ]] || false
+    # found
+    run dolt sql -q "SELECT * FROM twopk WHERE v2 = 61 AND v1 = 53"
+    [ "$status" -eq "0" ]
+    [[ "${lines[1]}" = "| pk1 | pk2 | v1 | v2 |" ]] || false
+    [[ "${lines[3]}" = "| 5   | 77  | 53 | 61 |" ]] || false
+    [[ "${lines[4]}" = "+-----+-----+----+----+" ]] || false
+    # not found key 1
+    run dolt sql -q "SELECT * FROM twopk WHERE v2 = 111 AND v1 = 53"
+    [ "$status" -eq "0" ]
+    [[ "${lines[1]}" = "| pk1 | pk2 | v1 | v2 |" ]] || false
+    [[ "${lines[3]}" = "+-----+-----+----+----+" ]] || false
+    # not found key 2
+    run dolt sql -q "SELECT * FROM twopk WHERE v2 = 61 AND v1 = 111"
+    [ "$status" -eq "0" ]
+    [[ "${lines[1]}" = "| pk1 | pk2 | v1 | v2 |" ]] || false
+    [[ "${lines[3]}" = "+-----+-----+----+----+" ]] || false
+    # not found key mismatch
+    run dolt sql -q "SELECT * FROM twopk WHERE v2 = 61 AND v1 = 54"
+    [ "$status" -eq "0" ]
+    [[ "${lines[1]}" = "| pk1 | pk2 | v1 | v2 |" ]] || false
+    [[ "${lines[3]}" = "+-----+-----+----+----+" ]] || false
+    # not indexed & found
+    run dolt sql -q "SELECT * FROM twopk WHERE v1 = 53"
+    [ "$status" -eq "0" ]
+    [[ "${lines[1]}" = "| pk1 | pk2 | v1 | v2 |" ]] || false
+    [[ "${lines[3]}" = "| 5   | 77  | 53 | 61 |" ]] || false
+    [[ "${lines[4]}" = "+-----+-----+----+----+" ]] || false
+    # not indexed & not found
+    run dolt sql -q "SELECT * FROM twopk WHERE v1 = 111"
+    [ "$status" -eq "0" ]
+    [[ "${lines[1]}" = "| pk1 | pk2 | v1 | v2 |" ]] || false
+    [[ "${lines[3]}" = "+-----+-----+----+----+" ]] || false
+}
+
+@test "index: SELECT > Full Match" {
+    skip "comparison does not yet use indexing"

```

Comment:
These are just functional tests that should work independent of indexing. I would unskip them. Then if we break them with a bad index implementation, we'll know.